plugin-directorymenu/directorymenu.cpp

```diff
@@ -0,0 +1,169 @@
+
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * END_COMMON_COPYRIGHT_HEADER */
+
+#include <vector>
+
+#include "directorymenu.h"
+#include <QDebug>
+#include <QDesktopServices>
+#include <QFileInfo>
+#include <QUrl>
+#include <QProcess>
+
+#include <XdgIcon>
+
+DirectoryMenu::DirectoryMenu(const ILxQtPanelPluginStartupInfo &startupInfo) :
+    QObject(),
+    ILxQtPanelPlugin(startupInfo),
+    mMenu(0)
+{
+    mOpenDirectorySignalMapper = new QSignalMapper(this);
+    mOpenTerminalSignalMapper = new QSignalMapper(this);
+    mMenuSignalMapper = new QSignalMapper(this);
+
+    mButton.setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+    mButton.setIcon(XdgIcon::fromTheme("folder"));
+    mButton.installEventFilter(this);
+
+    connect(&mButton, SIGNAL(clicked()), this, SLOT(showMenu()));
+    connect(mOpenDirectorySignalMapper, SIGNAL(mapped(QString)), this, SLOT(openDirectory(QString)));
+    connect(mOpenTerminalSignalMapper, SIGNAL(mapped(QString)), this, SLOT(openTerminal(QString)));
+    connect(mMenuSignalMapper, SIGNAL(mapped(QString)), this, SLOT(addMenu(QString)));
+}
+
+DirectoryMenu::~DirectoryMenu()
+{
+	if(mMenu)
+	{
+		delete mMenu;
+		mMenu = 0;
+	}
+}
+
+void DirectoryMenu::showMenu()
+{
+    buildMenu(QDir::homePath());
+
+    int x=0, y=0;
+
+    switch (panel()->position())
+    {
+        case ILxQtPanel::PositionTop:
+            x = mButton.mapToGlobal(QPoint(0, 0)).x();
+            y = panel()->globalGometry().bottom();
+            break;
+
+        case ILxQtPanel::PositionBottom:
+            x = mButton.mapToGlobal(QPoint(0, 0)).x();
+            y = panel()->globalGometry().top() - mMenu->sizeHint().height();
+            break;
+
+        case ILxQtPanel::PositionLeft:
+            x = panel()->globalGometry().right();
+            y = mButton.mapToGlobal(QPoint(0, 0)).y();
+            break;
+
+        case ILxQtPanel::PositionRight:
+            x = panel()->globalGometry().left() - mMenu->sizeHint().width();
+            y = mButton.mapToGlobal(QPoint(0, 0)).y();
+            break;
+    }
+
+    // Just using Qt`s activateWindow() won't work on some WMs like Kwin.
+    // Solution is to execute menu 1ms later using timer
+    mButton.activateWindow();
+    mMenu->exec(QPoint(x, y));
+}
+
+void DirectoryMenu::buildMenu(const QString& path)
+{
+	if(mMenu)
+	{
+		delete mMenu;
+		mMenu = 0;
+	}
+
+	mPathStrings.clear();
+
+	mMenu = new QMenu();
+
+	addActions(mMenu, path);
+}
+
+void DirectoryMenu::openDirectory(const QString& path)
+{
+	QDesktopServices::openUrl(QUrl("file://" + QDir::toNativeSeparators(path)));
+}
+
+void DirectoryMenu::openTerminal(const QString& path)
+{
+	QString command = QString("exo-open --working-directory \"%1\" --launch TerminalEmulator").arg(path);
```

Comment:
That was a question in my original post. exo-open is indeed from Xfce. The $TERM method does not work since it does not hold the default terminal name but instead the current terminal name [TERM](https://www.gnu.org/software/gettext/manual/html_node/The-TERM-variable.html)

Lxqt needs something equivalent to exo-open to make this work